raiden/network/rpc/client.py

```diff
@@ -541,27 +492,26 @@ def send_transaction(
                 'behavior.'
             )
 
-        if to == b'0' * 40:
+        if to == b'0' * 20:
             warnings.warn('For contract creation the empty string must be used.')
 
-        nonce = self.nonce()
-        startgas = self.check_startgas(startgas)
-
-        tx = Transaction(
-            nonce,
-            self.gasprice(),
-            startgas,
-            to=to,
+        transaction = dict(
+            nonce=self.nonce(),
+            gasPrice=self.gasprice(),
+            gas=self.check_startgas(startgas),
             value=value,
-            data=data,
+            data=data
         )
 
-        tx.sign(self.privkey)
-        result = self.rpccall_with_retry(
-            'eth_sendRawTransaction',
-            data_encoder(rlp.encode(tx)),
-        )
-        return result[2 if result.startswith('0x') else 0:]
+        # add the to address if not deploying a contract
+        if to != b'':
+            transaction['to'] = to_checksum_address(to)
+
+        signed_txn = self.web3.eth.account.signTransaction(transaction, self.privkey)
```

Comment:
isn't this call sending privkey over the wire? There are two possible attack vectors:

1) anyone who manages to shutdown the local ethereum node can start his own proxy (or wait for the original proxy restart) and MITM intercept the messages. In default geth configuration you don't need root access for that, anyone can bind port at `8545`.
2) there's no identity verification of the node you are connecting to and as far as I know the communication is unencrypted. Anyone who can listen to network traffic can see the privkey in plaintext.

Reply:
According to this docs this isn't a RPC call and should send the private key: https://eth-account.readthedocs.io/en/latest/eth_account.html#module-eth_account.account

Reply:
good! sometimes it's not really clear what's an actual RPC call